pkg/cloud/services/compute/instance.go

```diff
@@ -131,10 +141,6 @@ func (s *Service) constructNetworks(openStackCluster *infrav1.OpenStackCluster,
 	}
 	for i, port := range openStackMachine.Spec.Ports {
 		pOpts := &openStackMachine.Spec.Ports[i]
-		// No Trunk field specified for the port, inherit openStackMachine.Spec.Trunk.
-		if pOpts.Trunk == nil {
-			pOpts.Trunk = &openStackMachine.Spec.Trunk
-		}
```

Comment:
This logic has moved, and is now evaluated against *all* nets, not just ports.

Does this mean we now get trunks for networks where previously we only got them for ports if Trunk is set on the machine spec? Is that a good thing?

Reply:
We are still operating at the port level. Afaik, the logic hasn't changed in that regard.
```
		port := net.PortOpts
		if port != nil {
			// No Trunk field specified for the port, inherit openStackMachine.Spec.Trunk.
			if port.Trunk == nil {
				*port.Trunk = openStackMachine.Spec.Trunk
```